src/visitors/global_var_visitor.cpp

```diff
@@ -0,0 +1,65 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include <iostream>
+#include <memory>
+
+#include "ast/global.hpp"
+#include "ast/neuron_block.hpp"
+#include "ast/program.hpp"
+#include "ast/range.hpp"
+#include "ast/statement_block.hpp"
+#include "visitors/global_var_visitor.hpp"
+
+namespace nmodl {
+namespace visitor {
+
+void GlobalToRangeVisitor::visit_neuron_block(ast::NeuronBlock& node) {
+    ast::RangeVarVector range_variables;
+    std::vector<std::string> global_variables_to_remove;
+
+    auto& statement_block = node.get_statement_block();
```

Comment:
I'm wondering why you are doing everything ont he level of the neuron_block, rather than using the visitor to descend into the tree and do the work at the nodes themselves? It looks like you're not really using the visitor pattern..

Reply:
I am visiting the `NeuronBlock` instead of the `Global` or `GlobalVar` nodes since some of them would need to be deleted and this would become more complicated in this case, needing probably another pass on the AST the way I think about it. We can discuss maybe if you have a better idea on how this could be done

Reply:
I'm not insisting to change - we can keep it "for now" but I think that with the recently introduced `parent` pointers, this should be easier.

Reply:
Given your comment I started thinking this a bit more and added also the deletion of the `Global` statements if they are empty here https://github.com/BlueBrain/nmodl/pull/230/commits/5715a027f8ded426f7eb8e2d92e656ff3d2ce451. I believe this is the only issue with visiting the `Global` statements instead of the `NeuronBlock`, which is something that cannot be done easily I believe..

Reply:
> I'm wondering why you are doing everything ont he level of the neuron_block, rather than using the visitor to descend into the tree and do the work at the nodes themselves? It looks like you're not really using the visitor pattern..

to me it does not make much sense to do the job in the global nodes. It is difficult to know if you need to be deleted or not. It is your parent that has a clear view from up there and can decide who should live and who should convert. I  think @iomaganaris 's solution is fine

Reply:
> I'm not insisting to change - we can keep it "for now" but I think that with the recently introduced `parent` pointers, this should be easier.

even with parent is a lot of back and forth: you go down to the global node, we come back to delete myself, then, where do I go? I see a lot of pitfalls in deleting nodes when we are visiting them

Reply:
Based on above conversation, I am marking this as resolve then.

Reply:
After reconsidering this, I agree that for deletion it's better to delete from above, it does make sense.